.github/workflows/main.yml

```
@@ -118,26 +132,16 @@ jobs:
         verbose: true
 
   deploy:
+    needs: [check-package]
     if: github.event_name == 'push' && startsWith(github.event.ref, 'refs/tags') && github.repository == 'pytest-dev/pluggy'
     runs-on: ubuntu-latest
     permissions:
       id-token: write
     steps:
-      - uses: actions/checkout@v4
-        with:
-          fetch-depth: 0
-
-      - uses: actions/setup-python@v5
+      - name: Download built packages from the build-package job.
+        uses: actions/download-artifact@v4
         with:
-          python-version: "3.11"
-
-      - name: Install dependencies
-        run: |
-          python -m pip install --upgrade pip
-          pip install --upgrade build
-
-      - name: Build package
-        run: python -m build --sdist --wheel --outdir dist/
-
+          name: Packages
+          path: dist
       - name: Publish package
         uses: pypa/gh-action-pypi-publish@v1.9.0
```

While at it let's try the new attestations support:

```suggestion
        uses: pypa/gh-action-pypi-publish@v1.10.1
        with:
            attestations: true
```